### PR TITLE
Poetry plugin should update pyproject.toml

### LINKED ISSUE #2898: Make poetry version plugin to update the pyproject.toml as well not just the build
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
Currently the poetry plugin updates the version during the build, install phases but does not update the pyproject.toml
This makes it harder to easily decipher the version of the package.

**What did you expect to happen:**
Running poetry build or install should set the version of the package in pyproject.toml

### PR DESCRIPTION
**Pull request by andyi2it**
**What this PR does / why we need it**:
Makes poetry plugin to update pyproject.toml

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2898 

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

**Special notes for your reviewer**:

1. run `pip install python/plugin/poetry-version-plugin`
2. run `hack/python-release.sh`

s required, just write "NONE".
-->
```release-note
NONE
```

### CODE REVIEW

File: hack/python-release.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+cd ./python
+
+packages=$(find . -maxdepth 1 -type d)
+
+for folder in ${packages[@]}
+do
+    echo "moving into folder ${folder}"
+    if [[ ${folder} == 'plugin' || ! -f "${folder}/pyproject.toml" ]]; then
+        echo -e "\033[33mskipping folder ${folder}\033[0m"
```
Comment (yuzisun) [inline hack/python-release.sh:15]: What are these special characters ?

Comment (andyi2it) [reply hack/python-release.sh:15]: For adding yellow color as kind of warning that we are skipping these folders

File: hack/python-release.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+cd ./python
+
+packages=$(find . -maxdepth 1 -type d)
+
+for folder in ${packages[@]}
+do
+    echo "moving into folder ${folder}"
+    if [[ ${folder} == 'plugin' || ! -f "${folder}/pyproject.toml" ]]; then
```
Comment (yuzisun) [inline hack/python-release.sh:14]: note that this script requires installing the plugin with
`pip install ./python/plugin/poetry-version-plugin/`

**Pull request merged**